nlmod/dims/resample.py

```diff
@@ -431,7 +436,9 @@ def dim_to_regular_dim(da, dims, z):
         coords = dict(da.coords)
         coords["x"] = ds.x
         coords["y"] = ds.y
-        coords.pop("icell2d")
+        for key in list(coords):
```

Comment:
Why is this necessary? I think 'icell2d' is always in the coordinates of da because at the beginning of the function this checked.

Reply:
We need to delete all coordinates that have icell2d as a dimension, like 'xc', 'yc'. The coordinate 'icell2d' also has 'icell2d' as a dimension, so it is also removed. So no need to specifically delete it, this saves a line.

Reply:
Ah that makes sense, thanks for explaining